src/lib/hooks/swap/index.ts

```diff
@@ -45,3 +35,38 @@ export function useSwitchSwapCurrencies() {
     })
   }, [update])
 }
+
+export function useSwapCurrency(field: Field): [Currency | undefined, (currency?: Currency) => void] {
+  const atom = useMemo(() => pickAtom(swapAtom, field), [field])
+  const otherAtom = useMemo(() => pickAtom(swapAtom, otherField(field)), [field])
+  const [currency, setCurrency] = useAtom(atom)
+  const otherCurrency = useAtomValue(otherAtom)
+  const switchCurrencies = useSwitchCurrencies()
+  const setOrSwitchCurrency = useCallback(
+    (currency?: Currency) => {
+      if (currency === otherCurrency) {
+        switchCurrencies()
+      } else {
+        setCurrency(currency)
+      }
+    },
+    [otherCurrency, setCurrency, switchCurrencies]
+  )
+  return [currency, setOrSwitchCurrency]
+}
+
+export function useSwapAmount(field: Field): [string | undefined, (amount: string) => void] {
+  const amount = useAtomValue(amountAtom)
+  const independentField = useAtomValue(independentFieldAtom)
+  const value = useMemo(() => (independentField === field ? amount : undefined), [amount, independentField, field])
```

Comment:
Ahh i like having this reset to undefined (not sure if it actually did that before)